tests/test_campaigns.py

```diff
@@ -1,20 +1,172 @@
+            'erpc',
+            'institution',
+            'osf-preprints',
+            'socarxiv-preprints',
+            'engrxiv-preprints',
+            'psyarxiv-preprints',
+        ]
+        self.refresh = datetime.utcnow()
+        campaigns.CAMPAIGNS = None
+        campaigns.CAMPAIGNS_LAST_REFRESHED = self.refresh
+
+    def test_get_campaigns_init(self):
+        campaign_dict = campaigns.get_campaigns()
+        assert_equal(len(campaign_dict), len(self.campaign_lists))
+        for campaign in campaign_dict:
+            assert_in(campaign, self.campaign_lists)
+        assert_not_equal(self.refresh, campaigns.CAMPAIGNS_LAST_REFRESHED)
+
+    def test_get_campaigns_update_not_expired(self):
+        campaigns.get_campaigns()
+        self.refresh = campaigns.CAMPAIGNS_LAST_REFRESHED
+        campaigns.get_campaigns()
+        assert_equal(self.refresh, campaigns.CAMPAIGNS_LAST_REFRESHED)
+
+    def test_get_campaigns_update_expired(self):
+        campaigns.get_campaigns()
+        self.refresh = datetime.utcnow() - timedelta(minutes=5)
+        campaigns.CAMPAIGNS_LAST_REFRESHED = self.refresh
+        campaigns.get_campaigns()
+        assert_not_equal(self.refresh, campaigns.CAMPAIGNS_LAST_REFRESHED)
+
+
+# tests for campaign helper methods
+class TestCampaignMethods(OsfTestCase):
+
+    def setUp(self):
+        super(TestCampaignMethods, self).setUp()
+        set_preprint_providers()
+        self.campaign_lists = [
+            'prereg',
+            'erpc',
+            'institution',
+            'osf-preprints',
+            'socarxiv-preprints',
+            'engrxiv-preprints',
+            'psyarxiv-preprints',
+        ]
+        self.invalid_campaign = 'invalid_campaign'
+
+    def test_is_institution_login(self):
+        for campaign in self.campaign_lists:
+            institution = campaigns.is_institution_login(campaign)
+            if campaign == 'institution':
+                assert_true(institution)
+            else:
+                assert_false(institution)
+        institution = campaigns.is_institution_login(self.invalid_campaign)
+        assert_true(institution is None)
+
+    def test_is_native_login(self):
+        for campaign in self.campaign_lists:
+            native = campaigns.is_native_login(campaign)
+            if campaign == 'prereg' or campaign == 'erpc':
+                assert_true(native)
+            else:
+                assert_false(native)
+        native = campaigns.is_proxy_login(self.invalid_campaign)
+        assert_true(native is None)
+
+    def test_is_proxy_login(self):
+        for campaign in self.campaign_lists:
+            proxy = campaigns.is_proxy_login(campaign)
+            if campaign.endswith('-preprints'):
+                assert_true(proxy)
+            else:
+                assert_false(proxy)
+        proxy = campaigns.is_proxy_login(self.invalid_campaign)
+        assert_true(proxy is None)
+
+    def test_system_tag_for_campaign(self):
```

Comment:
For the next four tests, only test if the return value is `None` or `not None`. The `value` itself is not tested in order to avoid redundancy. Or we may want to initialize a test dictionary contains all detailed data and check the equality?